src/utils/labelUtils.ts

```diff
@@ -152,5 +152,12 @@ export const validateLabelKey: IValidationFunction = (key) => {
     };
   }
 
+  if (strKey.toLowerCase() === 'cluster.x-k8s.io/cluster-name') {
```

Comment:
Perhaps this deserves an explanatory comment or a link to this issue: https://github.com/giantswarm/giantswarm/issues/12431

For future code travellers that might come across it and wonder why.

Reply:
That said, the commit message will bring us here as well, so not super needed. But it's a nice touch.